src/core/ref-count.js

```diff
@@ -0,0 +1,33 @@
+// base class for reference counted objects
+class RefCount {
+    constructor() {
+        // counter
+        this._refCount = 0;
+    }
+
+    // inrements the counter
+    incRefCount() {
```

Comment:
Isn't it too much information in name of the methods?
`refCount.incRefCount()`?
Probably much more clean and "webby" would be: `refCount.increment()`.

Reply:
I would totally agree if this was a stand-alone class, but since it's been written as a base class, `increment` and `decrement` would be non-obvious for somebody reviewing subclass code that calls those functions.

Reply:
you call it as `mesh.incRefCount();` and not `refCount.incRefCount();`
Considering this is a base class, `mesh.increment();` would be too generic.

Reply:
Made a comment bellow, regarding it's purpose related to a class name.

Reply:
I called them incRef and decRef in the past, maybe that'd be a good middle ground